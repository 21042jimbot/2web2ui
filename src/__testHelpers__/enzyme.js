export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
