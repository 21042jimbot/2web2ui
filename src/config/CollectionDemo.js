import React from 'react';
import TableCollection from 'src/components/collection/TableCollection';
import Layout from 'src/components/layout/Layout';

// const headerComponent = () => <h1>Sci-Fi Movies</h1>;
// const rowComponent = ({ title, director, year, matchScore }) => (
//   <div>
//     <h3 style={{ marginBottom: '0' }}>{title} ({year})</h3>
//     <p>Directed by {director} {(typeof matchScore !== 'undefined') && <span>{` - Match Score: ${matchScore}`}</span>}</p>

//   </div>
// );
const props = {
  rows: [
    { title: 'Star Wars', director: 'George Lucas', year: 1977 },
    { title: 'Blade Runner', director: 'Ridley Scott', year: 1982 },
    { title: 'The Matrix', director: 'Lana and Lilly Wachowski', year: 1999 },
    { title: 'Terminator 2: Judgment Day', director: 'James Cameron', year: 1991 },
    { title: 'Jurassic Park', director: 'Steven Spielberg', year: 1993 },
    { title: 'Back to the Future', director: 'Robert Zemeckis', year: 1985 },
    { title: 'Mad Max: Fury Road', director: 'George Miller', year: 2015 },
    { title: '12 Monkeys', director: 'Terry Gilliam', year: 1995 },
    { title: 'Ex Machina', director: 'Alex Garland', year: 2015 },
    { title: 'Eternal Sunshine of the Spotless Mind', director: 'Michel Gondry', year: 2004 },
    { title: 'Looper', director: 'Rian Johnson', year: 2012 }
  ],
  columns: ['Title', 'Director', 'Released'],
  getRowData: (item) => [item.title, item.director, item.year],
  defaultPerPage: 5,
  perPageButtons: [3, 5, 10],
  pagination: true
};

export default function CollectionDemo() {
  return (
    <Layout.App>
      <TableCollection {...props} />
    </Layout.App>
  );
}
