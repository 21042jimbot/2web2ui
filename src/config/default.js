const config = {
  apiBase: 'no-default-set',
  apiDateFormat: 'YYYY-MM-DDTHH:mm',
  apiRequestTimeout: 60000,
  apiRequestHeaders: {
    'X-Sparky': '1d24c3473dd52a2f4a53fb6808cf9a73'
  },
  apiRequestBodyMaxSizeBytes: 500 * 1000, // ~ 500 KB
  authentication: {
    cookie: {
      name: 'auth',
      options: {
        path: '/'
      }
    },
    headers: {
      Authorization: 'Basic bXN5c1dlYlVJOmZhODZkNzJlLTYyODctNDUxMy1hZTdmLWVjOGM4ZmEwZDc2Ng==',
      'Content-Type': 'application/x-www-form-urlencoded'
    }
  },
  cardTypes: [
    { paymentFormat: 'visa', apiFormat: 'Visa' },
    { paymentFormat: 'mastercard', apiFormat: 'MasterCard' },
    { paymentFormat: 'amex', apiFormat: 'AmericanExpress' },
    { paymentFormat: 'discover', apiFormat: 'Discover' }
  ],
  chartColors: ['#04AEF9', '#fa6423', '#FFD300', '#8CCA3A', '#b94696'],
  cookieConsent: {
    cookie: {
      name: 'cookieConsent',
      ageDays: 365,
      options: {
        domain: 'sparkpost.com',
        path: '/'
      }
    }
  },
  featureFlags: {
    allow_default_signing_domains_for_ip_pools: false,
    allow_mailbox_verification: true,
    allow_anyone_at_verification: false,
    has_signup: false
  },
  heroku: {
    cookieName: 'heroku-nav-data'
  },
  maxUploadSizeBytes: 20000000,
  metricsPrecisionMap: [
    { time: 60, value: '1min', format: 'ha' },
    { time: 60 * 2, value: '5min', format: 'ha' },
    { time: 60 * 4, value: '15min', format: 'ha' },
    { time: 60 * 24 * 2, value: 'hour', format: 'ha' },
    { time: 60 * 24 * 7, value: 'day', format: 'MMM Do' },
    { time: 60 * 24 * 33, value: 'day', format: 'MMM Do' },
    { time: 60 * 24 * 190, value: 'week', format: 'MMM Do' },
    { time: Infinity, value: 'month', format: 'MMM YY' }
  ],
  release: process.env.REACT_APP_VERSION,
  sandbox: {
    localpart: 'sandbox',
    domain: 'sparkpostbox.com'
  },
  sendingIps: {
    maxPerAccount: 4,
    pricePerIp: 20.00,
    awsPricePerIp: 0.028
  },
  splashPage: '/reports/summary',
  summaryChart: {
    defaultMetrics: ['count_targeted', 'count_rendered', 'count_accepted', 'count_bounce']
  },
  support: {
    maxAttachmentSizeBytes: 500 * 1000 // ~ 500 KB
  },
  templates: {
    testData: {
      substitution_data: {},
      metadata: {},
      options: {}
    }
  },
  tenant: 'local',
  website: {
    domain: 'sparkpost.com'
  },
  zuora: {
    baseUrl: 'https://rest.apisandbox.zuora.com/v1',
    timeout: 15000
  },
  brightback: {
    baseUrl: 'https://app.brightback.com',
    app_id: '9N0rWBvKGR',
    save_return_url: '/account/billing', // Return URL from Brightback for end-users who do not cancel
    cancel_confirmation_url: '/accout/billing/plan?immediatePlanChange=free-0817', // Return URL from Brightback for end-users who cancel
    billing_url: '/account/billing/plan' // Billing URL to direct end-users to enter coupon code or other billing changes
  },
  smtpAuth: {
    host: 'no-default-set',
    port: 587,
    username: 'SMTP_Injection'
  },
  bounceDomains: {
    allowDefault: true,
    allowSubaccountDefault: false,
    cnameValue: 'sparkpostmail.com'
  },
  trackingDomains: {
    cnameValue: 'spgo.io'
  },
  dateFormat: 'MMM D YYYY',
  timeFormat: 'h:mma',
  messageEvents: {
    retentionPeriodDays: 10
  },
  recaptcha: {
    key: '6LeFZQETAAAAACWJfxw_DKHgEPnop3brlj9IsHrY',
    invisibleKey: '6LekChoUAAAAAJZouMPHnhRss2t7-ZetbAABfsOZ'
  },
  attribution: {
    cookieName: 'attribution',
    cookieDuration: 60 * 24 * 30,
    cookieDomain: '.sparkpost.com'
  },
  salesforceDataParams: ['src', 'utm_source', 'utm_medium', 'utm_campaign', 'utm_content', 'utm_term']
};

export default config;
