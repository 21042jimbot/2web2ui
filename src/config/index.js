import _ from 'lodash';
import defaultConfig from './default';
import envConfig from './env';

const mergedConfig = _.merge({}, defaultConfig, envConfig);

export default mergedConfig;
