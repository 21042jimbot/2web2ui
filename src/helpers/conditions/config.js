import _ from 'lodash';
import config from 'src/config';

export const configFlag = (path) => () => _.get(config, path, false);

export const configEquals = (path, value) => () => _.get(config, path) === value;

export const configNotEquals = (path, value) => () => _.get(config, path) !== value;
