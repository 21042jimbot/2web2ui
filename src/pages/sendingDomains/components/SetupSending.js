import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

// components
import { Panel, Grid, Banner, Button } from '@sparkpost/matchbox';
import { LongTextContainer } from 'src/components';
import SimpleTable from './SimpleTable';
import ReadyForIcon from './ReadyForIcon';
import VerifyEmail from './VerifyEmail';
import { SendingDomainSection } from './SendingDomainSection';
import { VerifiedIcon } from './Icons';

// actions
import { showAlert } from 'src/actions/globalAlert';
import { verifyDkim } from 'src/actions/sendingDomains';

import { resolveReadyFor } from 'src/helpers/domains';
import config from 'src/config';

export class SetupSending extends Component {
  state = {
    // verify via email modal
    open: false
  };

  showErrorAlert = ({ message }) => {
    const { domain: { id }, showAlert } = this.props;

    showAlert({ type: 'error', message: `Unable to verify DKIM record of ${id}. ${message}` });
  };

  verifyDomain = () => {
    const { domain: { id, subaccount_id: subaccount }, verifyDkim, showAlert } = this.props;

    return verifyDkim({ id, subaccount })
      .then((results) => {
        const readyFor = resolveReadyFor(results);
        if (readyFor.dkim) {
          showAlert({ type: 'success', message: `You have successfully verified DKIM record of ${id}` });
        } else {
          this.showErrorAlert({ message: results.dns.dkim_error });
        }
      }).catch((err) => {
        this.showErrorAlert({ message: err.message });
      });
  }

  renderText() {
    const readyFor = resolveReadyFor(this.props.domain.status);
    let content = null;

    if (!readyFor.sending && !readyFor.dkim) {
      content = <p><strong>To use this domain for sending</strong>, add this TXT record to your DNS settings, paying close attention to the specified hostname.</p>;

      // Append second paragraph for mailbox verification
      if (config.featureFlags.allow_mailbox_verification) {
        content = (
          <React.Fragment>
            {content}
            <p>
              We recommend DNS verification, but if you don't have DNS access, you
              can <UnstyledLink onClick={this.toggleVerifyViaEmailModal}> set this domain up for
              sending via email.</UnstyledLink>
            </p>
            {this.renderVerifyViaEmailModal()}
          </React.Fragment>
        );
      }
    }

    if (readyFor.sending && !readyFor.dkim) {
      content = <p>This domain is <strong>ready for sending</strong>, but if you plan to use it for sending, we still recommend that you add this TXT record to make it <strong>ready for DKIM</strong> as well.</p>;
    }

    return content;
  }

  getVerifyAction() {
    const { verifyDkimLoading } = this.props;
    const buttonText = verifyDkimLoading ? 'Verifying...' : 'Verify TXT Record';

    return {
      content: buttonText,
      onClick: this.verifyDomain,
      disabled: verifyDkimLoading
    };
  }

  toggleVerifyViaEmailModal = () => {
    const { open } = this.state;

    this.setState({
      open: !open
    });
  }

  renderVerifyViaEmailModal = () => {
    const { open } = this.state;
    const { domain } = this.props;

    if (!domain) {
      return null;
    }

    return (
      <VerifyEmail domain={domain} open={open} onCancel={this.toggleVerifyViaEmailModal}/>
    );
  }

  renderTxtRecordPanel() {
    const { domain: { dkimHostname, dkimValue, status }} = this.props;
    const readyFor = resolveReadyFor(status);

    if (readyFor.sending && readyFor.dkim) {
      return (
        <Panel sectioned>
          <p><VerifiedIcon/> This domain is all set up to send with DKIM-signing. Nice work!</p>
        </Panel>
      );
    }

    return (
      <Panel title='DNS Settings' sectioned actions={[this.getVerifyAction()]}>
        <LabelledValue label='Type'><p>TXT</p></LabelledValue>
        <LabelledValue label='Hostname'><p>{dkimHostname}</p></LabelledValue>
        <LabelledValue label='Value'><p>{dkimValue}</p></LabelledValue>
      </Panel>
    );
  }

  render() {
    return (
      <SendingDomainSection title="Set Up For Sending">
        <SendingDomainSection.Left>
          {this.renderText()}
        </SendingDomainSection.Left>
        <SendingDomainSection.Right>
          {this.renderTxtRecordPanel()}
        </SendingDomainSection.Right>
      </SendingDomainSection>
    );
  }
}

const mapStateToProps = ({ sendingDomains: { verifyDkimError, verifyDkimLoading }}) => ({
  verifyDkimError,
  verifyDkimLoading
});

export default withRouter(connect(mapStateToProps, { verifyDkim, showAlert })(SetupSending));
