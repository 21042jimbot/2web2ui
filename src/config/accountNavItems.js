import { OpenInNew, ExitToApp } from '@sparkpost/matchbox-icons';
import { LINKS } from 'src/constants';
import { isHeroku } from 'src/helpers/conditions/user';
import not from 'src/helpers/conditions/not';

export default [
  {
    label: 'Profile',
    to: '/account/profile',
    section: 1
  },
  {
    label: 'Billing',
    to: '/account/billing',
    section: 1
  },
  {
    label: 'Manage Users',
    to: '/account/users',
    section: 1
  },
  {
    label: 'API Documentation',
    external: true,
    to: LINKS.DOCS,
    icon: OpenInNew,
    section: 2,
    condition: not(isHeroku)
  },
  {
    label: 'Log Out',
    to: '/logout',
    icon: ExitToApp,
    section: 2,
    condition: not(isHeroku)
  }
];
