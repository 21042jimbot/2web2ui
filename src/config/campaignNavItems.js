
import { hasUiOption } from 'src/helpers/conditions/account';
import { FilterNone } from '@sparkpost/matchbox-icons/matchbox-icons';

const campaignNavItems = [
  {
    label: 'Templates',
    to: '/templates'
  },
  {
    label: 'A/B Testing',
    to: '/ab-testing'
  },
  {
    label: 'Snippets',
    to: '/snippets',
    tag: 'labs'
  }
];

export default {
  label: 'Campaigns',
  to: '/campaigns',
  icon: FilterNone,
  condition: hasUiOption('templatesV2'),
  children: campaignNavItems
};

