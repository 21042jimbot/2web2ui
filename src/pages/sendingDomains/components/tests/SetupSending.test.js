import { shallow, mount } from 'enzyme';
import React from 'react';
import config from 'src/config';

import { SetupSending } from '../SetupSending';

describe('Component: SetupSending', () => {
  let wrapper;
  let props;
  let status;

  beforeEach(() => {
    jest.resetModules();

    status = {
      ownership_verified: false,
      dkim_status: null
    };
    props = {
      domain: {
        id: 'xyz.com',
        subaccount_id: 999,
        dkimHostname: 'scph0118._domainkey.xyz.com',
        dkimValue: 'v=DKIM1; k=rsa; h=sha256; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQC7J/mQAVP9pynvD79opV/GD495pJWTWmHaEMymBO55y52gu2D23/CHbLys0KXszTAPOVS9x3HdRByeE1jIx41sZkWwDzxcyZH2NP5Mw3zNQWO4CqSGqrVPuukjIy5N3jmZnJ/V3IOkKnC3A+j76iIG42cVOgzoekbM3ZXSfwag+wIDAQAB',
        status
      },
      verifyLoading: false,
      verify: jest.fn(() => Promise.resolve()),
      showAlert: jest.fn()
    };

    config.featureFlags.allow_mailbox_verification = true;
    wrapper = shallow(<SetupSending {...props}/>);
  });

  it('renders correctly for unverified DKIM and invalid ownership with mailbox verification enabled', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when ownership and DKIM are verified', () => {
    const testProps = props.domain.status = { dkim_status: 'valid', ownership_verified: true };
    wrapper.setProps(testProps);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly for unverified DKIM and invalid ownership with mailbox verification disabled', () => {
    config.featureFlags.allow_mailbox_verification = false;
    const wrapper = shallow(<SetupSending {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when ownership is valid and DKIM is unverified', () => {
    props.domain.status = { dkim_status: 'invalid', ownership_verified: true };
    wrapper.setProps(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('disables the verification button while submitting', async() => {
    wrapper.setProps({ verifyLoading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should call updateTrackingDomain on submit', () => {
    const wrapper = mount(<SetupSending {...props}/>);

    wrapper.find('Button').simulate('click');
    expect(props.verify).toHaveBeenCalledTimes(1);
  });

  describe('verifyDomain', () => {
    let instance;
    beforeEach(() => {
      instance = wrapper.instance();
    });

    it('verifies domain and alerts when verification successful', async() => {
      props.verify.mockReturnValue(Promise.resolve({ dkim_status: 'valid' }));
      await instance.verifyDomain();
      expect(props.verify).toHaveBeenCalledTimes(1);
      expect(props.verify).toHaveBeenCalledWith({ id: 'xyz.com', type: 'dkim', subaccount: 999 });
      expect(props.showAlert).toHaveBeenCalledTimes(1);
      expect(props.showAlert).toHaveBeenCalledWith({ type: 'success', message: 'You have successfully verified DKIM record of xyz.com' });
    });

    it('alerts error when verification req is successful but verification is failed', async() => {
      props.verify.mockReturnValue(Promise.resolve({ dkim_status: 'invalid', dns: { dkim_error: 'nope!' }}));
      await instance.verifyDomain();
      expect(props.verify).toHaveBeenCalledTimes(1);
      expect(props.verify).toHaveBeenCalledWith({ id: 'xyz.com', type: 'dkim', subaccount: 999 });
      expect(props.showAlert).toHaveBeenCalledTimes(1);
      expect(props.showAlert).toHaveBeenCalledWith({ type: 'error', message: 'Unable to verify DKIM record of xyz.com. nope!' });
    });

    it('alerts when request is errored', async() => {
      const err = new Error('Request failed!');
      props.verify.mockReturnValue(Promise.reject(err));
      await instance.verifyDomain();
      expect(props.verify).toHaveBeenCalledTimes(1);
      expect(props.verify).toHaveBeenCalledWith({ id: 'xyz.com', type: 'dkim', subaccount: 999 });
      expect(props.showAlert).toHaveBeenCalledTimes(1);
      expect(props.showAlert).toHaveBeenCalledWith({ type: 'error', message: 'Unable to verify DKIM record of xyz.com. Request failed!' });
    });

  });
});
