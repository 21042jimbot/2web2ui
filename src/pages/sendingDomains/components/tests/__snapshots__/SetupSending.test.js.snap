// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SetupSending renders correctly for unverified DKIM and invalid ownership with mailbox verification disabled 1`] = `
<SendingDomainSection
  title="Set Up For Sending"
>
  <Left>
    <SubduedText
      as="p"
      color="gray.600"
      fontSize="300"
      mt="300"
    >
      <strong>
        To use this domain for sending
      </strong>
      , add this TXT record to your DNS settings, paying close attention to the specified hostname.
    </SubduedText>
  </Left>
  <Right>
    <SetupInstructionPanel
      isVerified={false}
      isVerifying={false}
      onVerify={[Function]}
      recordType="TXT"
      verifyButtonIdentifier="verify-dkim"
    >
      <LabelledValue
        label="Type"
      >
        <p>
          TXT
        </p>
      </LabelledValue>
      <LabelledValue
        label="Hostname"
      >
        <p>
          scph0118._domainkey.xyz.com
        </p>
      </LabelledValue>
      <LabelledValue
        label="Value"
      >
        <p>
          v=DKIM1; k=rsa; h=sha256; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQC7J/mQAVP9pynvD79opV/GD495pJWTWmHaEMymBO55y52gu2D23/CHbLys0KXszTAPOVS9x3HdRByeE1jIx41sZkWwDzxcyZH2NP5Mw3zNQWO4CqSGqrVPuukjIy5N3jmZnJ/V3IOkKnC3A+j76iIG42cVOgzoekbM3ZXSfwag+wIDAQAB
        </p>
      </LabelledValue>
    </SetupInstructionPanel>
  </Right>
</SendingDomainSection>
`;

exports[`Component: SetupSending renders correctly for unverified DKIM and invalid ownership with mailbox verification enabled 1`] = `
<SendingDomainSection
  title="Set Up For Sending"
>
  <Left>
    <SubduedText
      as="p"
      color="gray.600"
      fontSize="300"
      mt="300"
    >
      <strong>
        To use this domain for sending
      </strong>
      , add this TXT record to your DNS settings, paying close attention to the specified hostname.
    </SubduedText>
    <SubduedText
      as="p"
      color="gray.600"
      fontSize="300"
      mt="300"
    >
      We recommend DNS verification, but if you don't have DNS access, you can
       
      <UnstyledLink
        id="verify-with-email"
        onClick={[Function]}
      >
         
        set this domain up for sending via email.
      </UnstyledLink>
    </SubduedText>
  </Left>
  <Right>
    <SetupInstructionPanel
      isVerified={false}
      isVerifying={false}
      onVerify={[Function]}
      recordType="TXT"
      verifyButtonIdentifier="verify-dkim"
    >
      <LabelledValue
        label="Type"
      >
        <p>
          TXT
        </p>
      </LabelledValue>
      <LabelledValue
        label="Hostname"
      >
        <p>
          scph0118._domainkey.xyz.com
        </p>
      </LabelledValue>
      <LabelledValue
        label="Value"
      >
        <p>
          v=DKIM1; k=rsa; h=sha256; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQC7J/mQAVP9pynvD79opV/GD495pJWTWmHaEMymBO55y52gu2D23/CHbLys0KXszTAPOVS9x3HdRByeE1jIx41sZkWwDzxcyZH2NP5Mw3zNQWO4CqSGqrVPuukjIy5N3jmZnJ/V3IOkKnC3A+j76iIG42cVOgzoekbM3ZXSfwag+wIDAQAB
        </p>
      </LabelledValue>
    </SetupInstructionPanel>
  </Right>
</SendingDomainSection>
`;

exports[`Component: SetupSending renders correctly when ownership and DKIM are verified 1`] = `
<SendingDomainSection
  title="Set Up For Sending"
>
  <Left />
  <Right>
    <SetupInstructionPanel
      isVerified={true}
      isVerifying={false}
      onVerify={[Function]}
      recordType="TXT"
      verifyButtonIdentifier="verify-dkim"
    >
      <LabelledValue
        label="Type"
      >
        <p>
          TXT
        </p>
      </LabelledValue>
      <LabelledValue
        label="Hostname"
      >
        <p>
          scph0118._domainkey.xyz.com
        </p>
      </LabelledValue>
      <LabelledValue
        label="Value"
      >
        <p>
          v=DKIM1; k=rsa; h=sha256; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQC7J/mQAVP9pynvD79opV/GD495pJWTWmHaEMymBO55y52gu2D23/CHbLys0KXszTAPOVS9x3HdRByeE1jIx41sZkWwDzxcyZH2NP5Mw3zNQWO4CqSGqrVPuukjIy5N3jmZnJ/V3IOkKnC3A+j76iIG42cVOgzoekbM3ZXSfwag+wIDAQAB
        </p>
      </LabelledValue>
    </SetupInstructionPanel>
  </Right>
</SendingDomainSection>
`;

exports[`Component: SetupSending renders correctly when ownership is valid and DKIM is unverified 1`] = `
<SendingDomainSection
  title="Set Up For Sending"
>
  <Left>
    <SubduedText
      as="p"
      color="gray.600"
      fontSize="300"
      mt="300"
    >
      This domain is 
      <strong>
        ready for sending
      </strong>
      , but if you plan to use it for sending, we still recommend that you add this TXT record to make it 
      <strong>
        ready for DKIM
      </strong>
       
      as well.
    </SubduedText>
  </Left>
  <Right>
    <SetupInstructionPanel
      isVerified={false}
      isVerifying={false}
      onVerify={[Function]}
      recordType="TXT"
      verifyButtonIdentifier="verify-dkim"
    >
      <LabelledValue
        label="Type"
      >
        <p>
          TXT
        </p>
      </LabelledValue>
      <LabelledValue
        label="Hostname"
      >
        <p>
          scph0118._domainkey.xyz.com
        </p>
      </LabelledValue>
      <LabelledValue
        label="Value"
      >
        <p>
          v=DKIM1; k=rsa; h=sha256; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQC7J/mQAVP9pynvD79opV/GD495pJWTWmHaEMymBO55y52gu2D23/CHbLys0KXszTAPOVS9x3HdRByeE1jIx41sZkWwDzxcyZH2NP5Mw3zNQWO4CqSGqrVPuukjIy5N3jmZnJ/V3IOkKnC3A+j76iIG42cVOgzoekbM3ZXSfwag+wIDAQAB
        </p>
      </LabelledValue>
    </SetupInstructionPanel>
  </Right>
</SendingDomainSection>
`;
